### ISSUE TITLE
The avg losses are not same for training and validation when the same data are used.

### ISSUE THREAD

**Issue by p1n0cch10** [PaddlePaddle/models#4973]
Hi,

I am training a AttentionCluster for video classification by using my own data, followed here
https://github.com/PaddlePaddle/models/blob/release/1.8/PaddleCV/video/models/attention_cluster/README.md

I used the same data for training, validation, testing, and inference. In the training phrase, the avg losses for training and validation are different. The avg loss for training is about 0.002, but that for validation is 0.1.

My config file is something like this.

MODEL:
    name: "AttentionCluster"
    dataset: "YouTube-8M"
    bone_network: None
    drop_rate: 0.5 
    feature_num: 2
    feature_names: ['rgb', 'audio']
    feature_dims: [1024, 128]
    seg_num: 100 
    cluster_nums: [32, 32] 
    num_classes: 12
    topk: 20
    UNIQUE:
        good: 20
        bad: 30

TRAIN:
    epoch: 100 
    learning_rate: 0.001
    pretrain_base: None
    #batch_size: 2048
    batch_size: 128 
    use_gpu: True
    #num_gpus: 8
    num_gpus: 1
    filelist: "data/dataset/youtube8m/train.list"

VALID:
    #batch_size: 2048
    batch_size: 128 
    filelist: "data/dataset/youtube8m/train.list"

TEST:
    #batch_size: 256
    batch_size: 128 
    filelist: "data/dataset/youtube8m/train.list"

INFER:
    batch_size: 1
    filelist: "data/dataset/youtube8m/train.list"

Here is a piece of log of the last epoch.
[INFO: train_utils.py:   46]: ------- learning rate [0.001], learning rate counter [-] ----- 
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:49] Epoch 99, iter 0, time 0.34092283248901367,  , loss = 0.001345, Hit@1 = 0.40, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:49] Epoch 99, iter 1, time 0.18329644203186035,  , loss = 0.028296, Hit@1 = 0.41, PERR = 0.99, GAP = 1.00
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:49] Epoch 99, iter 2, time 0.1722731590270996,  , loss = 0.001181, Hit@1 = 0.47, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:50] Epoch 99, iter 3, time 0.17226123809814453,  , loss = 0.002051, Hit@1 = 0.43, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:50] Epoch 99, iter 4, time 0.17180728912353516,  , loss = 0.001143, Hit@1 = 0.42, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:50] Epoch 99, iter 5, time 0.17211627960205078,  , loss = 0.002029, Hit@1 = 0.45, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:50] Epoch 99, iter 6, time 0.171461820602417,  , loss = 0.002691, Hit@1 = 0.46, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TRAIN 2020-11-25 00:59:50] Epoch 99, iter 7, time 0.1721813678741455,  , loss = 0.001076, Hit@1 = 0.44, PERR = 1.00, GAP = 1.00
[INFO: train_utils.py:  122]: [TRAIN] Epoch 99 training finished, average time: 0.17362822805132186
[INFO: metrics_util.py:   79]: [TEST] test_iter 0  , loss = 0.474687, Hit@1 = 0.76, PERR = 0.98, GAP = 0.98
[INFO: metrics_util.py:   79]: [TEST] test_iter 1  , loss = 0.015227, Hit@1 = 1.00, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TEST] test_iter 2  , loss = 0.113151, Hit@1 = 0.98, PERR = 0.98, GAP = 1.00
[INFO: metrics_util.py:   79]: [TEST] test_iter 3  , loss = 0.117705, Hit@1 = 0.99, PERR = 0.99, GAP = 0.99
[INFO: metrics_util.py:   79]: [TEST] test_iter 4  , loss = 0.105678, Hit@1 = 0.46, PERR = 0.98, GAP = 1.00
[INFO: metrics_util.py:   79]: [TEST] test_iter 5  , loss = 0.105103, Hit@1 = 0.98, PERR = 0.99, GAP = 1.00
[INFO: metrics_util.py:   79]: [TEST] test_iter 6  , loss = 0.004947, Hit@1 = 1.00, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:   79]: [TEST] test_iter 7  , loss = 0.013899, Hit@1 = 1.00, PERR = 1.00, GAP = 1.00
[INFO: metrics_util.py:  112]: [TEST] Finish    avg_hit_at_one: 0.896331787109375,  avg_perr: 0.9912109375, avg_loss :0.1187995703658089,   aps: [0.9961657036415631, 0, 0, 0, 0, 0, 1.0000000000000002, 0, 0.9999999999999999, 1.0, 0, 0.9981928818589554],    gap:0.9960663751614023
share_vars_from is set, scope is ignored.

**Comment by huangjun12** [comment]
Data is randomly selected when training, so the avg_loss for training and validation may not same even you use the same datalist. Code in
https://github.com/PaddlePaddle/models/blob/d0e40f2e0e94788cffa746e9a75e4518510d2da3/PaddleCV/video/reader/feature_reader.py#L55

**Comment by p1n0cch10** [comment]
Thx. Could you please help me with the issue here ? I did not got any reply for 10 days. Many thanks.
https://github.com/PaddlePaddle/models/issues/4972#issue-749834627
